### PR TITLE
Update meta.yaml build/run environment for 4.0

### PR DESCRIPTION
**Pull request by krlberry**
##  Description
Update build and run environment in meta.yaml for 4.0 release. 

## Related Issue
ISIS 4.0 Release Candidate prep.

## Motivation and Context
Update build and run environment in meta.yaml for 4.0 release. 

## How Has This Been Tested?
Local build successfully built and linked some executables. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by scsides — DISMISSED** [review]
Reasonable

File: recipe/meta.yaml [hunk 1]
```diff
@@ -44,122 +44,115 @@ build:
...
   - qwt=6.1.3
-  - setuptools=38.5.1
-  - sip==4.18
-  - sqlite==3.13.0
-  - suitesparse==4.5.4
-  - superlu==5.2.1
-  - tnt==126=0
-  - wheel==0.30.0
-  - x264==20131218
-  - xalan-c==1.11
```
Comment (jessemapel) [inline recipe/meta.yaml:98]: I thought we still needed this for the build?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -44,122 +44,115 @@ build:
...
+  - ale >=0.7.1
+  - boost=1.68.0
+  - armadillo
+  - blas
+  - bullet=2.86.1=0
+  - bz2file
+  - bzip2
+  - cspice
+  - curl
+  - doxygen
```
Comment (jessemapel) [inline recipe/meta.yaml]: This can be removed

File: recipe/meta.yaml [hunk 3]
```diff
@@ -44,122 +44,115 @@ build:
...
+  - nanoflann
+  - nlohmann_json
   - conda-forge/label/gcc7::nn
-  - openblas==0.2.19
-  - opencv
-  - openssl==1.0.2n
-  - pcl==1.8.1
-  - pip==9.0.1
-  - protobuf==3.5.2
-  - python==3.6
```
Comment (jessemapel) [inline recipe/meta.yaml:143]: We may need to keep this in. I'm not 100% sure, needs testing.

File: recipe/meta.yaml (hunk 1 above)
Comment (scsides) [reply recipe/meta.yaml:98]: Only for building the docs. Xalan is the app, xerces is the lib

File: recipe/meta.yaml (hunk 2 above)
Comment (scsides) [reply recipe/meta.yaml]: agreed, same as Xalan

File: recipe/meta.yaml (hunk 3 above)
Comment (krlberry) [reply recipe/meta.yaml:143]: I'll run local tests on this without it in and report back (this will take some time.)

Comment (krlberry) [reply recipe/meta.yaml:143]: Update: Didn't even get to the stage of running local tests. Building this, pushing it up to my personal channel, installing it locally, and running any app results in "cant find opencv library" type errors. (Currently investigating)

Comment (krlberry) [reply recipe/meta.yaml:143]: @jessemapel We have `conda create -n isis3 python=3.6` as one of the first steps in our "install ISIS" instructions. Does this effectively accomplish the same thing?

Comment (jessemapel) [reply recipe/meta.yaml:143]: yes

**Review by scsides — APPROVED** [review]

**Pull request merged**